tests/jobs/test_backfill_job.py

```diff
@@ -71,14 +71,14 @@ def set_instance_attrs(self, dag_bag):
         self.parser = cli_parser.get_parser()
         self.dagbag = dag_bag
 
-    def _get_dummy_dag(self, dag_id, pool=Pool.DEFAULT_POOL_NAME, task_concurrency=None):
-        dag = DAG(dag_id=dag_id, start_date=DEFAULT_DATE, schedule_interval='@daily')
+    def _get_dummy_dag(
+        self, dag_maker, dag_id, pool=Pool.DEFAULT_POOL_NAME, task_concurrency=None, task_id='op', **kwargs
+    ):
 
-        with dag:
-            DummyOperator(task_id='op', pool=pool, task_concurrency=task_concurrency, dag=dag)
+        with dag_maker(dag_id=dag_id, schedule_interval='@daily', **kwargs) as dag:
+            DummyOperator(task_id=task_id, pool=pool, task_concurrency=task_concurrency)
 
-        dag.clear()
-        return dag
+        return dag, dag_maker.dag_run
```

Comment:
It might be possible to further simply the code by also turning this into a fixture.

```python
@pytest.fixture(scope="the_same_as_dag_maker")
def get_dummy_dag_and_run(dag_maker):
    def _get_dummy_dag_and_run(
        dag_id,
        pool=Pool.DEFAULT_POOL_NAME,
        task_concurrency=None,
        task_id='op',
        **kwargs,
    ):
        with dag_maker(dag_id=dag_id, schedule_interval='@daily', **kwargs) as dag:
            DummyOperator(task_id=task_id, pool=pool, task_concurrency=task_concurrency)
        return dag, dag_maker.dag_run

    return _get_dummy_dag_and_run
```

And use it like

```python
def test_unfinished_dag_runs_set_to_failed(self, get_dummy_dag_and_run):
    dag, dag_run = get_dummy_dag_and_run(dag_id='dummy_dag')